### PR TITLE
Prevent headless exception when deleting a node via the API

### PR DESCRIPTION
**Pull request by psiinon**
Moved purge method from PopupMenuPurgeSites to ExtensionHistory - not ideal but couldnt think of a better place for now.

### CODE REVIEW

**Comment by alienatix** [comment]
Any ideas when it will be merged?

**Comment by psiinon** [comment]
When its been oked by @thc202 and @kingthorin ;)

**Comment by alienatix** [comment]
🙇

**Comment by kingthorin** [comment]
Looks good

File: src/org/zaproxy/zap/extension/history/PopupMenuPurgeHistory.java [hunk 1]
```diff
@@ -97,7 +97,13 @@ private void purgeHistory(HistoryReference ref) {
 
         if (node.getHistoryReference() == ref) {
             // same active Node
-            PopupMenuPurgeSites.purge(map, node);
+            ExtensionHistory extHistory =
+                    Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class);
+
+            if (extHistory != null) {
+                extHistory.purge(map, node);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/history/PopupMenuPurgeHistory.java]: Should use the instance variable "extension" instead (the null check can also be removed).

File: src/org/zaproxy/zap/extension/history/PopupMenuPurgeSites.java [hunk 2]
```diff
@@ -70,64 +65,12 @@ public void performHistoryReferenceActions(List<HistoryReference> hrefs) {
 
     @Override
     public void performAction(SiteNode sn) {
-        purge(Model.getSingleton().getSession().getSiteTree(), sn);
-    }
-
-    public static void purge(SiteMap map, SiteNode node) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/history/PopupMenuPurgeSites.java]: We should keep (and deprecate) this method, to keep binary compatibility.

File: src/org/zaproxy/zap/extension/stdmenus/PopupExcludeFromProxyMenu.java [hunk 3]
```diff
@@ -68,7 +70,14 @@ public void performAction(SiteNode sn) {
 			session.getExcludeFromProxyRegexs().add(new StructuralSiteNode(sn).getRegexPattern());
 			SiteMap map = (SiteMap) View.getSingleton().getSiteTreePanel().getTreeSite().getModel();
 
-			PopupMenuPurgeSites.purge(map, sn);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/stdmenus/PopupExcludeFromProxyMenu.java]: The import of PopupMenuPurgeSites should be removed.

**Comment by thc202** [comment]
There's a merge commit that should be removed. (The branch could be rebased on latest develop too.)

**Comment by psiinon** [comment]
Updated with those fixes

**Comment by thc202** [comment]
Looks good to me, thank you!

**Comment by cwjcsu** [comment]
waiting for this merge

**Comment by kingthorin** [comment]
> kingthorin merged commit 2005d89 into zaproxy:develop 19 days ago

**Pull request merged**